### PR TITLE
Add analysis of moran process

### PR DESCRIPTION
**Pull request by drvinceknight**

### CODE REVIEW

File: main.tex [hunk 1]
```diff
@@ -500,6 +500,80 @@ \section{Results}\label{section:results}
...
+In a population of \(N\) total individuals of two types, the probability that \(k\)
+individuals of a given type can take over the population, the fixation
+probability \(x_k\) is given by:
+
+\[ x_k = \frac{ 1 + \sum_{j=1}^{k-1}\prod_{i=1}^j\gamma_i }{ 1 +
+\sum_{j=1}^{N-1}\prod_{i=1}^j\gamma_i } \]
+
+where:
+
+\[ \gamma_i = \frac{ p_{k, k - 1} }{ p_{k, k + 1} } \]
```
Comment (Nikoleta-v3) [inline main.tex]:
```suggestion
\[ \gamma_i = \frac{ p_{k, k - 1} }{ p_{k, k + 1} }. \]
```

File: main.tex [hunk 2]
```diff
@@ -500,6 +500,80 @@ \section{Results}\label{section:results}
...
+
+In the case considered here \(k\) is the number of best response players in the
+population, so that \(N - k\) is the number of opponents.
+
+The difference proposed here is to allow the best response player to act
```
Comment (Nikoleta-v3) [inline main.tex]: Different to what?

File: main.tex [hunk 3]
```diff
@@ -500,6 +500,80 @@ \section{Results}\label{section:results}
...
+probability \(x_k\) is given by:
+
+\[ x_k = \frac{ 1 + \sum_{j=1}^{k-1}\prod_{i=1}^j\gamma_i }{ 1 +
+\sum_{j=1}^{N-1}\prod_{i=1}^j\gamma_i } \]
+
+where:
+
+\[ \gamma_i = \frac{ p_{k, k - 1} }{ p_{k, k + 1} } \]
+
+In the case considered here \(k\) is the number of best response players in the
```
Comment (Nikoleta-v3) [inline main.tex]: Just making sure: we do mean small k here right?

File: main.tex [hunk 4]
```diff
@@ -500,6 +500,80 @@ \section{Results}\label{section:results}
...
+using the theory of mind to find the best response to not only the opponent but
+also the distribution of the population.
+Thus for every value of \(k\) there is a different best response player.
+
+In the case of the best response player the transition probabilities depend on the payoff matrix \(A ^ {(k)}\) where:
+
+\(A ^ {(k)}_{11}\) is the long run utility of the best response player against itself.
+\(A ^ {(k)}_{12}\) is the long run utility of the best response player against the opponent.
+\(A ^ {(k)}_{11}\) is the long run utility of the opponent against the best response player.
+\(A ^ {(k)}_{11}\) is the long run utility of the opponent against itself.
```
Comment (Nikoleta-v3) [inline main.tex]:
```suggestion

\begin{itemize}
\item \(A ^ {(k)}_{11}\) is the long run utility of the best response player against itself.
\item \(A ^ {(k)}_{12}\) is the long run utility of the best response player against the opponent.
\item \(A ^ {(k)}_{11}\) is the long run utility of the opponent against the best response player.
\item \(A ^ {(k)}_{11}\) is the long run utility of the opponent against itself.
\end{itemize}
```

File: main.tex [hunk 5]
```diff
@@ -500,6 +500,80 @@ \section{Results}\label{section:results}
...
+
+The matrix \(A ^ {(k)}\) is calculated using (\ref{eq:optimisation_quadratic})
```
Comment (Nikoleta-v3) [inline main.tex]:
```suggestion
The matrix \(A ^ {(k)}\) is calculated using Eq. (\ref{eq:optimisation_quadratic})
```

File: main.tex [hunk 6]
```diff
@@ -500,6 +500,80 @@ \section{Results}\label{section:results}
...
+for each value of \(k\) once the best response dynamics algorithm has calculated
```
Comment (Nikoleta-v3) [inline main.tex]: Can we maybe describe this with a bit better

File: main.tex [hunk 7]
```diff
@@ -500,6 +500,80 @@ \section{Results}\label{section:results}
...
+the best response player.
+
+Using this, the total utilities/fitnesses for each player can be written down:
+
+\[f_1^{(k)} = (k - 1) A_{11}^{(k)} + (N - k)A_{12}^{(k)}\]
+
+\[f_2^{(k)} = (k) A_{21}^{(k)} + (N - k - 1)A_{22}^{(k)}\]
+
+(\(f_1^{(k)}\) is the fitness of the best response player, and \(f_2^{(k)}\) is
+the fitness of the opponent)
```
Comment (Nikoleta-v3) [inline main.tex]:
```suggestion
where \(f_1^{(k)}\) is the fitness of the best response player, and \(f_2^{(k)}\) is
the fitness of the opponent.
```

File: main.tex [hunk 8]
```diff
@@ -500,6 +500,80 @@ \section{Results}\label{section:results}
...
+which are all that are required to calculate \(x_k\).
+
+Figure~\ref{fig:dynamic_moran_process_results} shows the results of an analysis
+of \(x_k\) for dynamically updating players. This is obtained over %TODO Add count of data points here
+data points. The ratio shows a relatively high performance compared to a non
+dynamic best response strategy the mean \(x_K\) over all values of \(K\) and all
+experiments is given by % TODO Add mean
+and in some cases this dynamic updating results in a 25\% increase in the
+absorption probability.
+As denoted before it is clear that the best response strategy does not have a
```
Comment (Nikoleta-v3) [inline main.tex]: long sentence. I think this paragraph needs a bit more. Maybe even explain the ratio here as well as in the caption.

**Review by Nikoleta-v3 — COMMENTED** [review]
I reviewed the addition on the Moran results.

Regarding the manuscript as a whole I believe it needs restructure and a bit more discussion on the conclusions of the Moran process.

File: main.tex (hunk 2 above)
Comment (drvinceknight) [reply main.tex]: Yeah this paragraph needs to be improved it.

File: main.tex (hunk 3 above)
Comment (drvinceknight) [reply main.tex]: No we mean big `K` throughout.

File: main.tex (hunk 6 above)
Comment (drvinceknight) [inline main.tex]: Needs to be big `\(K\)`

Comment (drvinceknight) [reply main.tex]: Yeah this can all be improved, we could have a picture of the Moran process and a small numeric example.

File: main.tex (hunk 8 above)
Comment (drvinceknight) [reply main.tex]: Yeah we could possibly define some notation for the non dynamic case and explain that.

**Comment by drvinceknight** [comment]
268edc2 updates the figures with latest moran data (as of 2020-09-11).

Link to the figure: https://github.com/Nikoleta-v3/Memory-size-in-the-prisoners-dilemma/blob/268edc22e2c21b07e87b0153ab1594b789348e9a/img/dynamic_moran_process_results/main.pdf

**Comment by Nikoleta-v3** [comment]
I am going to go ahead an merge this `PR` so I can compile and review the manuscript with all the changes (including #77 ).

**Comment by Nikoleta-v3** [comment]
We can `PR` our final suggestions to `prep-manuscript-for-resubmission`

**Pull request merged**